lib/rubocop/cop/layout/line_length.rb

```diff
@@ -132,6 +132,7 @@ def check_for_breakable_semicolons(processed_source)
 
         def check_for_breakable_block(block_node)
           return unless block_node.single_line?
+          return if block_node.each_ancestor(:def).any?(&:endless?)
```

Comment:
`:defs` should also be covered in the same way. Can you add test cases and update the implementation?